core/sql/sqlcomp/CmpSeabaseDDLview.cpp

```diff
@@ -1498,32 +1577,50 @@ PrivColumnBitmap colGrantableBitmap;
       }
       int32_t usedColNumber = naCol->getPosition();
      
-      // Grab privileges from the NATable structure
-      PrivMgrUserPrivs *privs = naTable->getPrivInfo();
-      if (privs == NULL) 
+      PrivMgrUserPrivs privs;
+      PrivMgrUserPrivs *pPrivInfo = NULL;
+      if (viewCreator)
+        pPrivInfo = naTable->getPrivInfo();
+      else
+      {
+        PrivStatus retcode = privInterface.getPrivileges((int64_t)naTable->objectUid().get_value(),
+                                                           naTable->getObjectType(),
+                                                           naTable->getOwner(),
+                                                           privs);
+
+        if (retcode == STATUS_ERROR)
+        {
+           *CmpCommon::diags() << DgSqlCode(-CAT_UNABLE_TO_RETRIEVE_PRIVS);
+           return false;
+        }
+        pPrivInfo = &privs;
```

Comment:
So, in one code path pPrivInfo points to a heap-allocated structure while in this code path it points to a stack variable. Is there any issue with memory leaks in the first case?